File: LearningJava/Polymorphism/CommissionEmployee.java
public class CommissionEmployee extends Employee
{
  private double grossSales;
  private double commissionRate;
  
  public CommissionEmployee(String firstName, String lastName, String socialSecurityNumber, 
          double grossSales, double commissionRate)
  {
    super(firstName, lastName, socialSecurityNumber);
    
    this.grossSales = grossSales;
    this.commissionRate = commissionRate;
  }
  
  @Override
  public double earnings()
  {
    return 0;
  }
  
  @Override
  public String toString()
  {
    return String.format("%sgrossSales = %.2f%ncommissionRate = %.2f%n", super.toString(), grossSales, commissionRate);
  }
       
}

File: C++Programming/dynamic_cast/polymorphism/CommissionEmployee.cpp
//
//  CommissionEmployee.cpp
//  polymorphism
//
//  Created by Vijay Kumar on 9/18/13.
//  Copyright (c) 2013 Vijay Kumar. All rights reserved.
//

#include "CommissionEmployee.h"
#include "Employee.h"
#include <iostream>
#include <string>

CommissionEmployee::CommissionEmployee(const string& first_name, const string& last_name,
                                       const string& ssn, const double gross_sales,
                                       const double commission_rate) 
:Employee(first_name, last_name, ssn)
{
  grossSales = gross_sales;
  commissionRate = commission_rate;
}

double CommissionEmployee::earnings() const
{
  return grossSales * commissionRate;
}

void CommissionEmployee::print() const
{
  cout << "Commission Employee: " << endl;
  Employee::print();
  cout << "GrossSales = " << grossSales << endl;
  cout << "CommissionRate = " << commissionRate << endl;
}